src/main/kotlin/org/stellar/walletsdk/utils/validateTransaction.kt

```diff
@@ -0,0 +1,24 @@
+package org.stellar.walletsdk.utils
+
+import org.stellar.sdk.Server
+import org.stellar.sdk.Transaction
+
+fun validateTransaction(transaction: Transaction, server: Server) {
+  val sourceAccount =
+    try {
+      server.accounts().account(transaction.sourceAccount)
+    } catch (e: Exception) {
+      throw Error("Source account ${transaction.sourceAccount} does not exist")
+    }
+
+  val sourceAccountBalance = accountAvailableNativeBalance(sourceAccount).toBigDecimal()
+  val transactionFees = stroopsToLumens(transaction.fee.toString()).toBigDecimal()
+
+  if (sourceAccountBalance < transactionFees) {
+    throw Error(
```

Comment:
Error is very generic. It is good to start defining Wallet SDK exceptions. Such as `InvalidArgumentException` so that the caller of the SDK can process the specific error and report to the customer. 

[`Error`](https://docs.oracle.com/javase/6/docs/api/java/lang/Error.html) is used to indicate serious problems that a reasonable caller should not try to catch. While `Exception` is the bread and butter for exception handling.

Reply:
Thanks for the explanation, Jamie! I'll replace all `Error` with `Exception`.